force-app/main/default/classes/SOQL.cls

```diff
@@ -1589,6 +1600,11 @@ public virtual inherited sharing class SOQL implements Queryable {
             groupByFields.add(field.getDescribe().getName());
         }
 
+        public void with(String relationshipName, SObjectField field) {
+            setGroupByFunction('{0}');
+            groupByFields.add(relationshipName + '.' + field.getDescribe().getName());
```

Comment:
We can do something like that, Apex invokes `.getDescribe().getName()` automatically.
It applies only when the String is concatenated with `SObjectField`.

`groupByFields.add(relationshipName + '.' + field);`